src/ActionBar/ActionBar.js

```diff
@@ -1,42 +1,121 @@
+    ...props
+}, ref) => {
+
     const actionBarClasses = classnames(
         'fd-action-bar',
         className
     );
 
     useEffect(() => {
         if (!disableStyles) {
-            require('fundamental-styles/dist/fonts.css');
             require('fundamental-styles/dist/action-bar.css');
         }
     }, []);
 
+    const actionBarHeaderClasses = classnames(
+        'fd-action-bar__header',
+        className
+    );
+
+    const actionBarBackClasses = classnames(
+        'fd-action-bar__back',
+        buttonContainerClassNames
+    );
+
+    const actionBarDescriptionClasses = classnames(
+        'fd-action-bar__description',
+        {
+            'fd-action-bar__description--back': onClick
+        }
+    );
+
+    const actionBarActionsClasses = classnames(
+        'fd-action-bar__actions',
+        actionClassNames
+    );
+
+    const HeadingTag = `h${headingLevel}`;
+
     return (
         <div {...props}
             className={actionBarClasses}
             ref={ref}>
-            {children}
+            <div {...props} className={actionBarHeaderClasses}>
+                {onClick && (<div className={actionBarBackClasses}>
+                    <Button
+                        {...buttonProps}
+                        compact
+                        disableStyles={disableStyles}
+                        glyph='navigation-left-arrow'
+                        onClick={onClick}
+                        option='transparent' />
+                </div>)}
+                <HeadingTag {...titleProps} className='fd-action-bar__title'>{title}</HeadingTag>
+                {actions && (
+                    <div {...actionProps} className={actionBarActionsClasses}>{actions}</div>
+                )}
+            </div>
+            {description &&
+                <p
+                    {...descriptionProps}
+                    className={actionBarDescriptionClasses}>{description}</p>
+            }
         </div>
     );
 });
 
 ActionBar.displayName = 'ActionBar';
 
 ActionBar.propTypes = {
-    children: PropTypes.node,
+    title: PropTypes.string.isRequired,
+    actionClassNames: PropTypes.string,
+    actionProps: PropTypes.object,
+    actions: PropTypes.node,
+    buttonContainerClassNames: PropTypes.string,
+    buttonProps: PropTypes.object,
     className: PropTypes.string,
-    disableStyles: PropTypes.bool
+    description: PropTypes.string,
+    descriptionProps: PropTypes.object,
+    disableStyles: PropTypes.bool,
+    headingLevel: CustomPropTypes.range(1, 6),
+    titleProps: PropTypes.object,
+    onClick: PropTypes.func
+};
+
+ActionBar.defaultProps = {
+    headingLevel: 3
+};
+
+ActionBar.propDescriptions = {
+    actionClassNames: 'Classnames to spread to the action Button container.',
+    actionProps: 'Props to spread to the action Button container',
+    actions: 'Button components to add to the ActionBar.',
+    buttonContainerClassNames: 'Classnames to spread to the back Button container.',
```

Comment:
Seems weird to have `actionClassNames` and `buttonContainerClassNames` plural when we use `className` which isn't plural for the same idea.

Reply:
Looks like there is a mixture of both ways of doing it in fundamental-react

```
actionClassNames
buttonContainerClassNames
labelClassNames
inputClassNames
listClassNames
tokenizerClassNames

vs

backdropClassName
contentClassName
inputClassName
popperClassName
tableBodyClassName
table....ClassName
referenceClassName
```

Reply:
I think that came from a bunch of different people implementing it. We should do a separate story to clean it up in one go. 

I'll make this one singular for now.

Reply:
https://github.com/SAP/fundamental-react/issues/911

Reply:
Now that I look at the history on it.... it's only me that's been adding the plural ones 😆🤣